python-api/tests/integrationtests/test_controller.py

```diff
@@ -239,15 +208,87 @@ def test_get_preview_ports(self):
                 sources.terminate_audio()
                 serv.terminate(1)
             finally:
-                if serv.proc:
-                    poll = serv.proc.poll()
-                    print self.__class__
-                    if poll == -11:
-                        print "SEGMENTATION FAULT OCCURRED"
-                    print "ERROR CODE - {0}".format(poll)
-                    serv.terminate(1)
-                    log = open('server.log')
-                    print log.read()
+                assert_no_segfault(serv)
+
+
+class TestSignals(object):
+
+    """Test on_*() methods"""
+
+    def __init__(self):
+        self.mainloop = GLib.MainLoop()
+        self.quit_count = 0
+
+    def run_mainloop(self):
+        """Start the MainLoop, set Quit-Counter to Zero"""
+        self.quit_count = 0
+        self.mainloop.run()
+
+    def quit_mainloop(self, *_):
```

Comment:
Yes, and `Mock(side_effect=self.quit_mainloop)` passes the arguments passed to `test_cb`on to the `side_effect`: http://www.voidspace.org.uk/python/mock/mock.html